src/export/ExportWavPack.cpp

```diff
@@ -0,0 +1,430 @@
+
+   int quality = gPrefs->Read(wxT("/FileFormats/WavPackEncodeQuality"), 1);
+   bool hybridMode = gPrefs->ReadBool(wxT("/FileFormats/WavPackHybridMode"), false);
+   bool createCorrectionFile = gPrefs->ReadBool(wxT("/FileFormats/WavPackCreateCorrectionFile"), false);
+   int bitRate = gPrefs->Read(wxT("/FileFormats/WavPackBitrate"), 160);
+
+   config.num_channels = numChannels;
+   config.sample_rate = rate;
+   config.channel_mask = config.num_channels == 1 ? 4 : 3; // Microsoft standard, mono = 4, stereo = 3
+   config.bits_per_sample = 16;
+   config.bytes_per_sample = 2;
+
+   if (quality == 0) {
+      config.flags |= CONFIG_FAST_FLAG;
+   } else if (quality == 1) {
+      config.flags |= CONFIG_HIGH_FLAG;
+   } else {
+      config.flags |= CONFIG_VERY_HIGH_FLAG;
+   }
+
+   if (hybridMode) {
+      config.flags |= CONFIG_HYBRID_FLAG;
+      config.flags |= CONFIG_BITRATE_KBPS;
+      config.bitrate = bitRate;
+
+      if (createCorrectionFile) {
+         config.flags |= CONFIG_CREATE_WVC;
+         outWvcFile.file = std::make_unique< FileIO >(fName.GetFullPath().Append("c"), FileIO::Output);
+      }
+   }
+
+   WavpackContext *wpc = WavpackOpenFileOutput(this->WriteBlock, &outWvFile, createCorrectionFile ? &outWvcFile : NULL);
+   if (!WavpackSetConfiguration64(wpc, &config, -1, NULL) || !WavpackPackInit(wpc)) {
+      WavpackCloseFile(wpc);
+      return ProgressResult::Failed;
+   }
+
+// Samples to write per run
+#define SAMPLES_PER_RUN 8192u
+
+   uint32_t bufferSize = SAMPLES_PER_RUN * numChannels;
+   ArrayOf<int32_t> wavpackBuffer{ bufferSize };
+   auto updateResult = ProgressResult::Success;
+   {
+      auto mixer = CreateMixer(tracks, selectionOnly,
+         t0, t1,
+         numChannels, SAMPLES_PER_RUN, true,
+         rate, int16Sample, mixerSpec);
+
+      InitProgress( pDialog, fName,
+         selectionOnly
+            ? XO("Exporting selected audio as Wavpack file")
+            : XO("Exporting the audio as Wavpack file") );
+      auto &progress = *pDialog;
+
+      while (updateResult == ProgressResult::Success) {
+         auto samplesThisRun = mixer->Process(SAMPLES_PER_RUN);
+
+         if (samplesThisRun == 0)
+            break;
+
+         char *mixed = (char *)(const char*)mixer->GetBuffer();
+         for (decltype(samplesThisRun) j = 0; j < samplesThisRun; j++) {
+            for (size_t i = 0; i < numChannels; i++) {
+               int32_t value = *mixed++ & 0xff;
+               value += *mixed++ << 8;
+               wavpackBuffer[j*numChannels + i] = value;
+            }
+         }
+
+         if (!WavpackPackSamples(wpc, wavpackBuffer.get(), samplesThisRun))
+            updateResult = ProgressResult::Cancelled;
+
+         if (updateResult == ProgressResult::Success)
+            updateResult =
+               progress.Update(mixer->MixGetCurrentTime() - t0, t1 - t0);
+      }
+   }
+
+   if (!WavpackFlushSamples(wpc)) {
+      updateResult = ProgressResult::Cancelled;
```

Comment:
`return ProgressResult::Failed`?

And I think that `BasicUI::ShowErrorDialog` should be used to notify the user about the error in places where codec has failed. 

(Please, see ExportPCM for example)